### PR TITLE
Stuff

### LINKED ISSUE #104: Update the CI
TODO:
- [ ] Fix URLS
- [ ] Fix Giles

Fixed Upstream:
Fix website build.

### LINKED ISSUE #109: Issue with Web navigation on cellphone
While I was on the sunpy.org website in developer tools, I switched the "Toggle Device" to IPhoneX to see what it would look like in that format.  When I switched, I noticed several issues: The drop down menu was a hidden in the top right corner (it is just a outlined box), the dimensions of the page were not formatted to the screen (so I had to zoom out to see the whole webpage), and the only submenu I could access from the drop-down menu was "SunPy Project".  I double checked this on my OnePlus device and the same issues occurred.

### LINKED ISSUE #111: Update CircleCI config

### LINKED ISSUE #112: Increase contrast of links in sidebar
If you put a link in a title you get this:

image

We probably want to set the link style to be the bright orange or something.

### LINKED ISSUE #115: Add new affiliated packages to the dropdown docs menu
aiapy and pfsspy recently gained affiliated package status but have not been added to the drop-down menu. I'm not sure about pfsspy, but aiapy would be happy to adopt the sunpy sphinx theme if it is added to the menu.

### LINKED ISSUE #116: Add Open Graph metadata to enable better embedding in external sites
Adding Open Graph metadata tags is extremely simple and will result in an improved appearance for sunpy.org links shared on sites such as Facebook and LinkedIn.

See section on "Improve content sharing with Open Graph" at https://developer.linkedin.com/docs/share-on-linkedin#

### PR DESCRIPTION
**Pull request by nabobalis**
Copies https://github.com/sunpy/sunpy/pull/4503 
Fixes #115
Fixes #116
Fixes #111 
Fixes #104
Fixes #112
Fixes #109

### CODE REVIEW

**Comment by Cadair** [comment]
I would imagine (but got lazy and didn't look into it) that we should be able to define toctree once and then inherit the base template to modify it. I can have a play later.

**Comment by nabobalis** [comment]
> I would imagine (but got lazy and didn't look into it) that we should be able to define toctree once and then inherit the base template to modify it. I can have a play later.

I will do this now.

**Comment by nabobalis** [comment]
Nope I was unable to do it. Probably missing something obvious.

File: sunpy_sphinx_theme/conf.py [hunk 1]
```diff
@@ -22,7 +22,11 @@ def page_url(page):
     return urljoin(sunpy_website_url_base, page)
 
 
-html_sidebars = {"**": ["docsidebar.html"]}
+html_sidebars = {
+    #  We use a custom sidebar for the gallery.
+    "generated/gallery/index": ["gallery_sidebar.html"],
```
Comment (Cadair) [inline sunpy_sphinx_theme/conf.py:27]: this wont pick up examples?

Comment (nabobalis) [reply sunpy_sphinx_theme/conf.py:27]: You mean if you are on the example?

File: pyproject.toml [hunk 2]
```diff
@@ -1,3 +1,44 @@
...
+    message = "Click details to preview the HTML documentation."
+    report_on_fail = true
+
+  [ tool.gilesbot.circleci_artifacts.website ]
+    url = "sunpy.org/_build/html/index.html"
+    message = "Click details to see the website build"
+    report_on_fail = true
+
+  [ tool.gilesbot.pull_requests ]
+    enabled = true
```
Comment (nabobalis) [inline pyproject.toml:44]: This doesn't work?

File: sunpy_sphinx_theme/sunpy/static/sunpy_style.css [hunk 3]
```diff
@@ -391,18 +391,17 @@ p.admonition-title {
```
Comment (nabobalis) [inline sunpy_sphinx_theme/sunpy/static/sunpy_style.css:391]: Some choice CSS changes that people might not like.

File: sunpy_sphinx_theme/conf.py (hunk 1 above)
Comment (nabobalis) [reply sunpy_sphinx_theme/conf.py:27]: You can have a look at the example docs, they arent as deep as the sunpy ones but should be enough for now.

File: sunpy_sphinx_theme/sunpy/static/sunpy_style.css [hunk 4]
```diff
@@ -824,3 +806,17 @@ table tr div.deprecated {
...
+.reference.internal.nav-header {
+  overflow-x: auto;
+}
+
+/*
+For any ``code`` on the sidebar.
+*/
+*[class^='docutils literal notranslate'] {
+  color: #fe7900;
+}
```
Comment (nabobalis) [inline sunpy_sphinx_theme/sunpy/static/sunpy_style.css:822]: I am not sure this actually works.

File: sunpy_sphinx_theme/conf.py (hunk 1 above)
Comment (nabobalis) [reply sunpy_sphinx_theme/conf.py]: Fixed.

File: sunpy_sphinx_theme/conf.py [hunk 5]
```diff
@@ -22,7 +22,16 @@ def page_url(page):
     return urljoin(sunpy_website_url_base, page)
 
 
-html_sidebars = {"**": ["docsidebar.html"]}
+html_sidebars = {
+    "*": ["docsidebar.html"],
```
Comment (nabobalis) [inline sunpy_sphinx_theme/conf.py:26]: This does enforce a style on the docs unless a package overwrites this.

File: .circleci/config.yml [hunk 6]
```diff
@@ -29,11 +30,29 @@ jobs:
...
-      - image: circleci/python:3.6
+      - image: circleci/python:3.8
+    steps:
+      - checkout
+      - run: *apt-install
+      - run: *pip-install
+      - run: pip install -U setuptools | cat
+      - run: pip install -e . | cat
+      - run: git clone https://github.com/sunpy/sunpy.git
+      - run: pip install -e sunpy/.[docs] | cat
```
Comment (nabobalis) [inline .circleci/config.yml:45]:
```suggestion
      - run: pip install -e sunpy/.[docs] | cat
      - run: python -c "import sunpy.data.sample"
```

File: sunpy_sphinx_theme/conf.py [hunk 7]
```diff
@@ -45,9 +54,12 @@ def page_url(page):
         (
             "Documentation",
             [
-                ("SunPy", "https://docs.sunpy.org/en/stable/", 1),
+                ("sunpy", "https://docs.sunpy.org/en/stable/", 1),
                 ("ndcube", "https://docs.sunpy.org/projects/ndcube/", 1),
                 ("drms", "https://docs.sunpy.org/projects/drms/", 1),
+                ("aiapy", "https://aiapy.readthedocs.io/en/latest/", 1),
```
Comment (wtbarnes) [inline sunpy_sphinx_theme/conf.py:60]:
```suggestion
                ("aiapy", "https://aiapy.readthedocs.io/en/stable/", 1),
```

**Comment by wtbarnes** [comment]
Also see [this merge request to aiapy](https://gitlab.com/LMSAL_HUB/aia_hub/aiapy/-/merge_requests/83) which adds the sunpy theme.

File: .circleci/config.yml [hunk 8]
```diff
@@ -29,11 +30,30 @@ jobs:
           path: sunpy.org/_build/html
       - run:
           name: "Built documentation is available at:"
-          command: DOCS_URL="${CIRCLE_BUILD_URL}/artifacts/${CIRCLE_NODE_INDEX}/${CIRCLE_WORKING_DIRECTORY/#\~/$HOME}/sunpy.org/_build/html/index.html"; echo $DOCS_URL
+          command: DOCS_URL="${CIRCLE_BUILD_URL}/artifacts/${CIRCLE_NODE_INDEX}/sunpy.org/_build/html/index.html"; echo $DOCS_URL
 
-  test_docs:
+  sunpy:
```
Comment (nabobalis) [inline .circleci/config.yml:35]: Need to fix this.

**Comment by nabobalis** [comment]
Sunpy doc build probably needs a patch to the main repo.

**Comment by nabobalis** [comment]
View on the iphone x for me now.

Untitled

**Comment by nabobalis** [comment]
Need to double check the open graph stuff

**Comment by nabobalis** [comment]
Broken the website build.

File: sunpy_sphinx_theme/sunpy/navbar.html [hunk 9]
```diff
@@ -11,7 +11,7 @@
       </button>
       <a class="navbar-brand" href="{{ theme_logo_url }}">
         {%- block sidebarlogo %}
-        {%- if logo %}<span><img src="{{ pathto('_static/' + logo, 1) }}"></span>{%- endif %}
+        {%- if logo %}<span><img src="{{ pathto('static/' + logo, 1) }}"></span>{%- endif %}
```
Comment (nabobalis) [inline sunpy_sphinx_theme/sunpy/navbar.html:14]: ?!

File: sunpy_sphinx_theme/sunpy/theme.conf [hunk 10]
```diff
@@ -10,7 +10,8 @@ bootstrap_version = 3
 
 # Defaults for SunPy projects
 logo_url = https://sunpy.org/
-opengraph_image = img/sunpy_icon.svg
+# Opengraph does not support SVGs
+opengraph_image = sunpy_icon_128x128.png
```
Comment (nabobalis) [inline sunpy_sphinx_theme/sunpy/theme.conf:14]: The issue is also that the path for this should be a full url which means adding this to the project's repo as well.

**Pull request merged**